Parser improvements for German Scene

JBou opened the pull request:
#### Database Migration
NO

#### Description

- Parse [UHD2BD](https://www.xrel.to/p2p/releases.html?search_results=5enI4xfOXR114028624&query=UHD2BD&maxed_out=0) as BluRay instead of HDTV
- Parsing of German Scene REMUX releases. This would be important for Germans, they have other [scene rules](https://scenerules.org/t.html?id=2016_DE_AVC.nfo). Without this, it's a lot of testing and searching for custom formats to make this work correctly, and newbies don't have enough knowledge to tell the difference between the releases, when they are all listed as BluRay-1080p. Radarr should correctly parse them out of the box.
#6307
I took the regex from the issue, but adjusted it to parse also German Remuxes without DL in the release title.

- Fixes remaining periods (or multiple spaces) when filtering SimpleReleaseTitle.
The SimpleReleaseTitle is used when parsing the title, year, episodes, ...
If there are 2 or more points in the title, this may cause unexpected results and makes it harder to debug.
Before the fix: The.Move.Title.Extended.German..BluRay.-RlsGrp
After the fix: The.Move.Title.Extended.German.BluRay-RlsGrp

- Correctly parse SCENE and P2P BluRay DISKS releases
#6307
Again, I took the regex from the issue and adjusted it to pass the tests that were already there.
It should work now and already have been tested with custom formats, but please have a look at it again.

- At last, I improved parsing of the release when the word "german", "french" or "truefrench" is in the title. It should further avoid false positives to close #6474
Whats new: when there is no year, it now only matches the word german when it is before a common scene word (BluRay, DL, HDTV, ...). It is also not matched, when the title starts with "German" or "German" is in the middle of the title.
The Only problematic titles left are the ones that end in "German", but I got all titles from themoviedb when you search for german, and there were only 3 titles with german at the end:
`The.German`
`Herman.the.German`
`Aleksei.German`
I manually excluded them in the regex, so we don't parse the language as German.
.
I just had to exclude the following tests, because they are not strict enough and could cause false positives for the english releases:
`Valana la Movie TRUEFRENCH BluRay 720p 2016 kjhlj`
`Der.Movie.James.German.Bluray.FuckYou.Pso.Why.cant.you.follow.scene.rules.1998`
I haven't found any releases in the format of the TRUEFRENCH one on my trackers/indexers and on predb. I hope we can exclude this test, otherwise it may become difficult or quite impossible. If someone of you has a better idea, please let me know.
For sure we can exclude the German Pso one, I [can't find any release in that format](https://www.xrel.to/group-PsO-release-list.html). It just makes things complicated, even if there are no releases in that format.

@Qstick, please have a look at it :)


#### Screenshot (if UI related)

#### Todos
- [x] Tests
- [ ] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #6307
* Fixes #6474

bakerboy448 commented:
> Valana la Movie TRUEFRENCH BluRay 720p 2016 kjhlj
> Der.Movie.James.German.Bluray.FuckYou.Pso.Why.cant.you.follow.scene.ules.1998
> I haven't found any releases in the format of the TRUEFRENCH

Tons of Results for TRUEFRENCH
![image](https://user-images.githubusercontent.com/55419169/151255661-e210933a-0a2f-497c-8c29-fd3ad4dacc1f.png)

Tons of Results on PReDB for TRUEFRENCH
![image](https://user-images.githubusercontent.com/55419169/151255805-5ecca6c2-b992-40c4-92d8-a8762224fc15.png)

bakerboy448 commented on src/NzbDrone.Core.Test/ParserTests/ParserFixture.cs:
```diff
@@ -23,7 +23,9 @@ public class ParserFixture : CoreTest
          * Constantine S1-E1-WEB-DL-1080p-NZBgeek
-         * [TestCase("Valana la Movie FRENCH BluRay 720p 2016 kjhlj", "Valana la Movie")]  Removed 2021-12-19 as this / the regex for this was breaking all movies w/ french in title
+         * Valana la Movie FRENCH BluRay 720p 2016 kjhlj
+         * Valana la Movie TRUEFRENCH BluRay 720p 2016 kjhlj
```
TRUEFRENCH test case cannot be removed; This is a valid release in the wild.

JBou replied:
Okay, we could keep the old regex for TRUEFRENCH and additionally use my new regex for FRENCH and GERMAN. Or I try to change the new regex to match also the TRUEFRENCH release above, but this would make it more complicated.
Do you have some more example for releases in this format?

JBou replied:
I missed your screenshots. The releases above should not match with my regex, because it's very strict. Therefore, when the language comes after the year, it will be matched be the [standard regex](https://github.com/Radarr/Radarr/pull/6999/files#diff-66eb7a94e4c0bcb08b7dd279734bda078d52ea70a99c0df3dcfaf13e86e0602fL27).

Tomorrow I will test the releases above to make sure.

JBou replied:
The problem with the TRUEFRENCH test above is, that the year is in the wrong position.

bakerboy448 replied:
from glancing through the ~600 results in movie cats that Prowlarr returned; it seems you're correct - does not appear there are instances with TRUEFRENCH before the name in the wild?

JBou replied:
Here are the Tests I used to update the regex: https://regex101.com/r/8uEUSr/2
As you can see, releases with a year in the title are not matched with this regex, they are then handled by the second regex (except titles where the Language tag comes before the year `Der.Film.deines.Lebens.German.2011.PAL.DVDR-ETM`)

I tested all the releases in the screenshots above, none of them match, which is the expected result.

ItsME6969 commented:
@JBou
For both `German` and `English` REMUXES this should suffice
`^(?=.*\b(Blu[-_. ]?ray|HD[-_. ]?DVD|BD|UHD)\b)(?=.*(\bREMUX\b|((?<=\d{4}).*German.*(DL)?)(?=.*\b(AVC|HEVC|VC[-_. ]?1|MVC|MPEG[-_. ]?2)))).*`
For BR-DISKS you used an older one and this should do for ALL SCENE & P2P
`^(?!.*\b((?<!HD[._ -]|HD)DVD|BDRip|720p|MKV|XviD|WMV|d3g|REMUX|[xh][-_. ]?26[45]|German[-_. ]?DL)\b)(((?=.*(Blu[-_. ]?ray|BD|HD[-_. ]?DVD)\b)(?=.*\b(AVC|HEVC|VC[-_. ]?1|MVC|MPEG[-_. ]?2|BDMV)\b))|^((?=.*\b(^((?=.*\b((.*_)?COMPLETE.*|Dis[ck])\b)(?=.*(Blu[-_. ]?ray|HD[-_. ]?DVD)))|3D[-_. ]?BD|BR[-_. ]?DISK|Full[-_. ]?Blu[-_. ]?ray|^((?=.*((BD|UHD)[-_. ]?(25|50|66|100)))(?=.*ISO)?))))).*`

@bakerboy448 the BR-Disk CF regex has been thoroughly tested for a long time now by @TRaSH-  and everybody who uses it and I'm confident it can replace the existing regex. All the test cases are already [here ](https://github.com/Radarr/Radarr/issues/6307)

JBou commented:
@Qstick can you please tell me how to proceed with this PR? :)

Qstick commented:
There some things in here if like to take a closer look at, just haven't had time

jaype87 commented:
While you're add it, could you have a look at the language detection? I think "German DL" should be interpreted as German and English, but it seems to only interpret it as German.

JBou commented:
> While you're add it, could you have a look at the language detection? I think "German DL" should be interpreted as German and English, but it seems to only interpret it as German.

I rebased the PR and added the fix for the problem with "German DL". Wasn't aware that it was broken.

I just noticed that one of the commits have already been merged: https://github.com/Radarr/Radarr/commit/a4136150d024e036580bae535e0fd0eea84c0a92

Qstick commented:
Being handled or already has been handled in various other PRs
